src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSJobManager.groovy

```diff
@@ -254,7 +254,8 @@ class PBSJobManager extends ClusterJobManager<PBSCommand> {
         GPathResult parsedJobs = new XmlSlurper().parseText(resultLines.last())
 
         parsedJobs.children().each { it ->
-            GenericJobInfo gj = new GenericJobInfo(it["Job_Name"] as String, null, it["Job_Id"] as String, null, it["depend"] ? (it["depend"] as  String).find("afterok.*")?.findAll(/(\d+).(\w+)/) { fullMatch, beforeDot, afterDot -> return beforeDot } : null)
+            String jobId = (it["Job_Id"] as String).split("\\.")[0]
```

Comment:
If there is no guarantee that [0] exists, I'd prefer some kind of parse error with information over a NullPointerException.

Furthermore this is ID splitting code that certainly already exists somewhere else. We had some discussion about this issue (short vs. long/full ID). Would you centralize the code and make it safe there?

Reply:
I think it would be better to use a BEJobId than just a string.

Reply:
Here you already change the semantics by splitting! It does not really matter whether you take BEJobID here or not, but the semanticts may (??) matter. Unfortunately, GenericJobInfo uses `id` instead of `jobId` to model the ID, so it is hard to figure out where the value is actually used.

I suggest you just change it and @dankwart-de and me fix the dependent code. Also consider changing the the `id` field of GenericJobInfo to `jobId`.